src/server/qgsserverogcapi.h

```diff
@@ -0,0 +1,203 @@
+ email                : [your-email-here]
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSERVEROGCAPI_H
+#define QGSSERVEROGCAPI_H
+
+#include "qgsserverapi.h"
+#include "qgis_server.h"
+
+
+class QgsServerOgcApiHandler;
+
+/**
+ * \ingroup server
+ * QGIS Server OGC API endpoint. QgsServerOgcApi provides the foundation for
+ * the new generation of REST-API based OGC services (e.g. WFS3).
+ *
+ * This class can be used directly and configured by registering handlers
+ * as instances of QgsServerOgcApiHandler.
+ *
+ * \code{.py}
+ *
+ * \endcode
+ *
+ * \since QGIS 3.10
+ */
+class SERVER_EXPORT QgsServerOgcApi : public QgsServerApi
+{
+
+    Q_GADGET
+
+  public:
+
+    // Note: non a scoped enum or qHash fails
+    //! Rel link types
+    enum Rel
+    {
+      // The following registered link relation types are used
+      alternate, //! Refers to a substitute for this context.
+      describedBy, //! Refers to a resource providing information about the link’s context.
+      collection, //! The target IRI points to a resource that is a member of the collection represented by the context IRI.
+      item, //! The target IRI points to a resource that is a member of the collection represented by the context IRI.
+      self, //! Conveys an identifier for the link’s context.
+      service_desc, //! Identifies service description for the context that is primarily intended for consumption by machines.
+      service_doc, //! Identifies service documentation for the context that is primarily intended for human consumption.
+      prev, //! Indicates that the link’s context is a part of a series, and that the previous in the series is the link targe
+      next, //! Indicates that the link’s context is a part of a series, and that the next in the series is the link target.
+      license, //! Refers to a license associated with this context.
+      // In addition the following link relation types are used for which no applicable registered link relation type could be identified:
+      items, //! Refers to a resource that is comprised of members of the collection represented by the link’s context.
+      conformance, //! The target IRI points to a resource which represents the collection resource for the context IRI.
+      data //! The target IRI points to resource data
+    };
+    Q_ENUM( Rel )
+
+    // Note: non a scoped enum or qHash fails
```

Comment:
```suggestion
    // Note: cannot be a scoped enum because qHash does not support them
```

If I read this correctly.